samples/trumpify/trumpify_review.py

```diff
@@ -1,106 +1,103 @@
         else:
-            print "@%s has not tweeted in the last %d minutes." % (INPUT_TWITTER_NAME, interval)
+            print "@%s has not tweeted in the last %d minutes." % (twitter_name, int(interval / MINUTES))
+
+        time.sleep(interval)
 
-        time.sleep(interval * 60)
 
+def translate_to_trump_version(message):
+    """
+    Create a Daemo task using "message" as data and pass it to workers for translation
 
-def post_to_daemo(message):
+    :param message: tweet object to be used as data for daemo task
+    """
     text = message.get('text')
     id = message.get('id')
 
-    client.publish(project_key=PROJECT_ID, tasks=[{
-        "id": id,
-        "tweet": text
-    }], approve=approve_tweet, completed=create_review_task, stream=True)
+    daemo.publish(
+        project_key=PROJECT_KEY,
+        tasks=[{
+            "id": id,
+            "tweet": text
+        }],
+        approve=approve_tweet,
+        completed=post_to_twitter
+    )
 
 
-def approve_tweet(results):
-    approvals = []
-    for result in results:
-        text = result.get('results')[0].get('result')
-        is_approved = len(text) > 0
-        approvals.append(is_approved)
-    return approvals
+def get_tweet_text(worker_response):
+    """
+    Filter out just the tweet text from a worker's complete submission
 
+    :param worker_response: submission made by a worker for a task
+    :return: actual tweet text
+    """
+    return worker_response.get('fields').get('tweet')
 
-def create_review_task(results):
-    tasks = [{
-                 "id": result.get('results')[0].get('id'),
-                 "tweet_result": result.get('results')[0].get('result')
-             } for result in results]
+def approve_tweet(worker_responses):
+    """
+    Verify each worker response if it meets the requirements
 
-    client.publish(
-        project_key=REVIEW_PROJECT_ID,
-        tasks=tasks,
-        approve=approve_review,
-        completed=post_to_twitter,
-        stream=True
-    )
+    :param worker_responses: submission made by a worker for a task
+    :return: list of True/False
+    """
+    approvals = [len(get_tweet_text(response)) > 0 for response in worker_responses]
+    return approvals
 
 
-def approve_review(results):
-    approvals = []
-    for result in results:
-        text = result.get('results')[0].get('result')
-        is_approved = len(text) > 0
-        approvals.append(is_approved)
-    return approvals
+def post_to_twitter(worker_responses):
+    """
+    Post worker response(s) to twitter and set up for peer review
 
+    :param worker_responses: submission made by a worker for a task
+    """
 
-def post_to_twitter(results):
-    for result in results:
-        text = result.get('task_data').get('tweet_result')
+    for approval_response in worker_responses:
+        text = approval_response.get('task_data').get('tweet_result')
         twitter.post(text)
 
+    daemo.peer_review(worker_responses)
```

Comment:
Because we don't support translating from Trueskill scores to Boomerang scores yet, `rate` has no use, so we don't have a callback for it. Once that is implemented I see two possibilities.

We ask for a callback to be provided to `peer_review` which takes the updated Trueskill scores of the workers being peer reviewed. In this callback the requester can call `rate` and/or do operations on the scores. In this case the mental model is that `peer_review` produces the scores and `rate` translates and saves them, similar to how non-peer-reviewed projects work.

Or once we receive the updated Trueskill scores, we internally call `rate` ourselves, translating and saving the scores. In this case the mental model is that `peer_review` is a replacement for both producing scores and calling `rate`.